### DISCUSSION TITLE
Relation to hazardous by probabl - is that a copycat project?

### DISCUSSION THREAD

**Discussion by fkiraly** [CamDavidsonPilon/lifelines#1617] [General]
@CamDavidsonPilon, may I ask what is going on with the `hazardous` library by the company probabl (French company close to some sklearn developers)?

The most advanced state is probably the "ICML 2024" branch
https://github.com/soda-inria/hazardous/tree/icml-2024
On the technical level, it seems like:

* it wraps `lifelines` without clear credit, see [example 1](https://github.com/soda-inria/hazardous/blob/icml-2024/hazardous/_ipcw.py), [example 2](https://github.com/soda-inria/hazardous/blob/icml-2024/hazardous/_aalan_johansen.py), and some other packages like [pycox](https://github.com/soda-inria/hazardous/blob/icml-2024/hazardous/_deep_hit.py)
* it duplicates some features from `lifelines`, relying also on heavy wrapping, see [metrics module](https://github.com/soda-inria/hazardous/tree/icml-2024/hazardous/metrics)
* it it incomplete, has not been released on `pypi`, has not been worked on in the public repository since ca half a year. It may be in stealth mode, although if current progress can be extrapolated, it wlil not be ready in a state comparable to `lifelines`, for one or two years.

On the publicity level:
* it is prominent on the probabl webpage, listed as one of the 8 major libraries that probabl is funding (see below)
* it is currently being advertised on linkedin
* the name of the branch "ICML 2024" seems to imply that there will be a major fesature at ICML 2024, July.
* Olivier Grisel and probabl are repeatedly placing survival modelling content on social networks

I find that odd, especially the statement that it is one of probabl's flagship projects, given the basically completely unready state of hazardous and how much it wraps `lifelines`.

Do you have an agreement with probabl?

probabl webpage: https://probabl.ai/open-source

Screenshot, retrieved June 9, 2024
![image](https://github.com/CamDavidsonPilon/lifelines/assets/7985502/34de7089-2dff-479c-b4ff-63ceb684926c)

**Comment by CamDavidsonPilon** [comment]
Hi @fkiraly, 

Thanks for bringing this to my attention. We don't have any connection to that project or company. While it would be nice to receiving trickle-down funding from a VC-backed company, I'm not expecting it. Nor am I expect expecting any attribution beyond import statements. 

What I do expect is that lifelines is a placeholder. These devs probably want to rewrite important models themselves (they have more experience with Cython and probably want control over internals) rather than defer out to lifelines. Probably they used lifelines-wrapped models for early demos.

**Reply by fkiraly** [reply]
Still, it looks like it is a demo for something that basically uses the same design as `lifelines`, but without acknowledgment.

If it is used in a funding pitch - as you say - then imo that is not very friendly, as you have spent a lot of time in `lifelines`, and someone else pitches for money with the same idea? This is an incomplete placeholder package since more than a year.

The ethical thing imo would be them funding `lifelines` (and/or `scikit-survival`) and rewarding those who spent a lot of time on it, instead of making their own copy and playing down design influences.

One would believe that is what they would be doing, given the values on their webpage (reference here: https://papers.probabl.ai/our-manifesto)

![image](https://github.com/CamDavidsonPilon/lifelines/assets/7985502/3d870bbf-b85c-4ae8-8aa7-85e370e5841b)

But, as you say, at the end of the day, they are VC funded, and perhaps it's only valid until it is "refined" into a better manifesto...

![image](https://github.com/CamDavidsonPilon/lifelines/assets/7985502/52a339d4-a970-4158-a534-16cf40bced23)
